### PR TITLE
updated Prometheus readme and added How Tos

### PR DESCRIPTION
**Pull request by lee-baines**
Filled out the documentation and made it clearer about the components involved. Should provide a good base for the documentation later.

### CODE REVIEW

File: helm/forgerock-metrics/README.md [hunk 1]
```diff
@@ -6,15 +6,25 @@ Alertmanager overview: [Overview](https://prometheus.io/docs/alerting/overview/)
...
-    * deploy the Prometheus Operator along with Grafana and Alert Manager and other Helm charts that help monitor GKE.
-    * connect to the Prometheus and Grafana endpoints.
-* forgerock-metrics Helm chart that provides configurable ServiceMonitors and a job to automatically import Grafana dashboards for ForgeRock products.  ServiceMonitors define the ForgeRock Identity Platform component endpoints that are monitored by Prometheus.
-* values files that are used by the deploy script and can be edited to customize the configuration of Prometheus, Grafana and Alert Manager.
-* auto-import folder which provides a Dockerfile for producing the docker image used by the import-dashboards job.
+**Prometheus solution comprises of the following artifacts:**  
+
+Helm charts:
+* ***prometheus-operator*** which creates custom resources which makes the Prometheus deployment native to Kubernetes and configuration through Kubernetes manifests.
+* ***kube-prometheus*** which contains multiple sub charts including Prometheus, Grafana and Alertmanager and other Helm charts that help monitor GKE.
```
Comment (dgoldssfo) [inline helm/forgerock-metrics/README.md]: s/sub charts/subcharts/
s/help monitor/monitor/

File: helm/forgerock-metrics/README.md [hunk 2]
```diff
@@ -23,19 +33,20 @@ Alertmanager configuration: [Config](https://prometheus.io/docs/alerting/configu
...
 
 # How Grafana works
 
-The Grafana Helm chart is deployed as part of the kube-prometheus chart.  This comes prepackaged with ServiceMonitors and  
-dashboards that monitor various aspects of the GKE cluster including the cluster node resources and Kubernetes objects.
+The Grafana Helm chart is deployed as part of the kube-prometheus chart.  Grafana automatically connects to Prometheus and syncs all  
+the metrics which are visible through Graphs.  
 
-Dashboards for ForgeRock products are imported into Grafana after Grafana has been deployed. 
+Dashboards for ForgeRock products are imported into Grafana post deployment. This is carried out by a Kubernetes job that mounts the dashboard files
```
Comment (dgoldssfo) [inline helm/forgerock-metrics/README.md]: s/post deployment/post-deployment/
s/This is carried out by a Kubernetes job that mounts/A Kubernetes job mounts/
Period at the end of this sentence, pls

File: helm/forgerock-metrics/README.md [hunk 3]
```diff
@@ -23,19 +33,20 @@ Alertmanager configuration: [Config](https://prometheus.io/docs/alerting/configu
...
 # How Grafana works
 
-The Grafana Helm chart is deployed as part of the kube-prometheus chart.  This comes prepackaged with ServiceMonitors and  
-dashboards that monitor various aspects of the GKE cluster including the cluster node resources and Kubernetes objects.
+The Grafana Helm chart is deployed as part of the kube-prometheus chart.  Grafana automatically connects to Prometheus and syncs all  
+the metrics which are visible through Graphs.  
 
-Dashboards for ForgeRock products are imported into Grafana after Grafana has been deployed. 
+Dashboards for ForgeRock products are imported into Grafana post deployment. This is carried out by a Kubernetes job that mounts the dashboard files  
+from helm/forgerock-metrics/dashboards folder, formats them to bypass a couple of limitations in the Grafana import API, and then uses the API to import them.
```
Comment (dgoldssfo) [inline helm/forgerock-metrics/README.md]: s/from/from the/

File: helm/forgerock-metrics/README.md [hunk 4]
```diff
@@ -57,10 +68,10 @@ A PrometheusRules CRD has been included in the Helm chart which includes the fr-
 * Deployed ForgeRock application in Google Cloud cluster.
 * Kubectl authenticated to cluster.
 * Separate namespace prepared for deploying Prometheus/Grafana
-(Use namespace that is different to where your application is running).
+(Use namespace that is different to where your application is running, default is 'monitoring').
```
Comment (dgoldssfo) [inline helm/forgerock-metrics/README.md]: s/Use namespace/Use a namespace/
s/running, default is 'monitoring'/running. The default is 'monitoring'./

File: helm/forgerock-metrics/README.md [hunk 5]
```diff
@@ -57,10 +68,10 @@ A PrometheusRules CRD has been included in the Helm chart which includes the fr-
...
 
 ### Prepare for deployment
-* cd to monitoring folder within forgeops repo.
+* cd to bin folder within forgeops repo.
```
Comment (dgoldssfo) [inline helm/forgerock-metrics/README.md]: s/to bin folder within forgeops repo/to the bin folder in your forgeops repo clone/

File: helm/forgerock-metrics/README.md [hunk 6]
```diff
@@ -76,7 +87,7 @@ Run the deploy script ./deploy_prometheus.sh with the OPTIONAL flags:
 
 ### View Prometheus/Grafana
 
-The following script uses kubectl port forwarding to access Prometheus and Grafana UIs. Run ./bin/connect_prometheus.sh with the following flags:
+The following script uses kubectl port forwarding to access Prometheus and Grafana UIs. Run ./connect_prometheus.sh with the following flags:
```
Comment (dgoldssfo) [inline helm/forgerock-metrics/README.md]: s/to access/to access the/

File: helm/forgerock-metrics/README.md [hunk 7]
```diff
@@ -134,12 +145,30 @@ If you want Prometheus to scrape metrics from a different product, you need to c
     ./deploy_prometheus.sh [-n <namespace>]
     ```
 
+### Configure alerting rules
+To add new alerting rules, add additional rules to fr-alerts.yaml. fr-alerts.yaml is split into groups with a group for each product and a  
+separate group for cluster rules.  
+
+See https://prometheus.io/docs/practices/alerting/ for details on configuring alerts.
```
Comment (dgoldssfo) [inline helm/forgerock-metrics/README.md]: You could add a link here if you feel like it

File: helm/forgerock-metrics/README.md [hunk 8]
```diff
@@ -134,12 +145,30 @@ If you want Prometheus to scrape metrics from a different product, you need to c
...
+separate group for cluster rules.  
+
+See https://prometheus.io/docs/practices/alerting/ for details on configuring alerts. 
+
+### Configure the alert message output(Slack).
+The alert output can be configured in the Alertmanager section of the kube-prometheus.yaml. In the slack_configs section of the receiver block,  
+you can define the template for the alert output.  The output text also incorporates labels so the info can be dynamically imported from the original  
+alert definition(see the Configuring alerting rules how to).  
+
+See https://prometheus.io/docs/alerting/configuration/ and https://prometheus.io/docs/alerting/notifications/ for more details.
```
Comment (dgoldssfo) [inline helm/forgerock-metrics/README.md]: And here

File: helm/forgerock-metrics/README.md [hunk 9]
```diff
@@ -134,12 +145,30 @@ If you want Prometheus to scrape metrics from a different product, you need to c
...
+
 
 ### Import Custom Grafana Dashboards
 
-The easiest way to import dashboards, is to manually import the json files in the GUI.
+The easiest way to import dashboards, is to manually import the json files in the GUI.
```
Comment (dgoldssfo) [inline helm/forgerock-metrics/README.md]: Pls remove the comma in the middle of this sentence.
Also s/json/JSON/

File: helm/forgerock-metrics/README.md [hunk 10]
```diff
@@ -134,12 +145,30 @@ If you want Prometheus to scrape metrics from a different product, you need to c
...
+alert definition(see the Configuring alerting rules how to).  
+
+See https://prometheus.io/docs/alerting/configuration/ and https://prometheus.io/docs/alerting/notifications/ for more details.
+
 
 ### Import Custom Grafana Dashboards
 
-The easiest way to import dashboards, is to manually import the json files in the GUI.
+The easiest way to import dashboards, is to manually import the json files in the GUI.  
 Currently exporting then importing dashboards via the HTTP api doesn't work correctly and requires manual amendments.
```
Comment (dgoldssfo) [inline helm/forgerock-metrics/README.md]: s/api/API/

**Comment by lee-baines** [comment]
Above suggested changes have been made. Thanks for spotting those

**Pull request merged**